### PR TITLE
add the python callstack for debug support test=develop

### PR DESCRIPTION
**Pull request by wopeizl**

### CODE REVIEW

File: paddle/fluid/framework/operator.cc [hunk 1]
```diff
@@ -19,17 +19,25 @@ limitations under the License. */
...
 #include "paddle/fluid/framework/var_type.h"
+#include "paddle/fluid/platform/debug_support.h"
 #include "paddle/fluid/platform/profiler.h"
 
 DECLARE_bool(benchmark);
 DEFINE_bool(check_nan_inf, false,
             "Checking whether operator produce NAN/INF or not. It will be "
             "extremely slow so please use this flag wisely.");
 
+DEFINE_bool(
```
Comment (panyx0718) [inline paddle/fluid/framework/operator.cc]: Can this be a config in executor run API?

Comment (panyx0718) [reply paddle/fluid/framework/operator.cc]: we want to support different debug levels. For level=1 print shape, for level=2 print debug NaN/Inf, etc

File: paddle/fluid/framework/operator.cc [hunk 2]
```diff
@@ -155,7 +163,20 @@ RuntimeContext::RuntimeContext(const VariableNameMap& innames,
   }
 }
 
+void OperatorBase::PreHook(const Scope& scope, const platform::Place& place) {
+  auto attrName = OpProtoAndCheckerMaker::OpCreationCallstackAttrName();
+  if (HasAttr(attrName)) {
+    auto& callstack = Attr<std::vector<std::string>>(attrName);
+    platform::PythonDebugSupport::GetInstance()->SetInformation(callstack);
```
Comment (panyx0718) [inline paddle/fluid/framework/operator.cc]: This python stack should be thrown all the time when there is an exception

File: paddle/fluid/framework/operator.h [hunk 3]
```diff
@@ -160,6 +160,10 @@ class OperatorBase {
                                  const platform::Place& place,
                                  const RuntimeContext& ctx) const {}
 
+  // Add the hooks
+  virtual void PreHook(const Scope& scope, const platform::Place& place);
```
Comment (panyx0718) [inline paddle/fluid/framework/operator.h]: No need for hook. Just print the debug information at the right code place.

**Comment by wopeizl** [comment]
image

File: paddle/fluid/framework/operator.h (hunk 3 above)
Comment (wopeizl) [reply paddle/fluid/framework/operator.h]: remove them

File: paddle/fluid/framework/operator.cc (hunk 2 above)
Comment (wopeizl) [reply paddle/fluid/framework/operator.cc]: ok

File: python/paddle/fluid/framework.py [hunk 4]
```diff
@@ -625,6 +626,11 @@ def __init__(self,
         if type is None:
             raise ValueError(
                 "`type` to initilized an Operator can not be None.")
+        else:
```
Comment (panyx0718) [inline python/paddle/fluid/framework.py:629]: seems no need for this else

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**